yt_dlp/extractor/digitalconcerthall.py

```diff
@@ -0,0 +1,156 @@
+    ExtractorError,
+    parse_resolution,
+    traverse_obj,
+    try_get,
+    urlencode_postdata,
+)
+
+
+class DigitalConcertHallIE(InfoExtractor):
+    IE_DESC = 'DigitalConcertHall extractor'
+    _VALID_URL = r'https?://(?:www\.)?digitalconcerthall\.com/(?P<language>[a-z]+)/concert/(?P<id>[0-9]+)'
+    _OAUTH_URL = 'https://api.digitalconcerthall.com/v2/oauth2/token'
+    _ACCESS_TOKEN = 'none'
+    _NETRC_MACHINE = 'digitalconcerthall'
+    # if you don't login, all you will get is trailers
+    _TESTS = [{
+        'url': 'https://www.digitalconcerthall.com/en/concert/53201',
+        'info_dict': {
+            'id': '53201-1',
+            'ext': 'mp4',
+            'title': 'Kurt Weill - [Magic Night]',
+            'thumbnail': r're:^https?://images.digitalconcerthall.com/cms/thumbnails.*\.jpg$',
+            'upload_date': '20210624',
+            'timestamp': 1624548600,
+        },
+        'params': {'skip_download': 'm3u8'},
+    }]
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if username is None:
+            raise ExtractorError('No login info available, needed for using %s.' % self.IE_NAME, expected=True)
+            return
+        token_response = self._download_json(
+            self._OAUTH_URL,
+            None, 'Obtaining token', errnote='Unable to obtain token', data=urlencode_postdata({
+                'affiliate': 'none',
+                'grant_type': 'device',
+                'device_vendor': 'unknown',
+                'device_model': 'unknown',
+                'app_id': 'dch.webapp',
+                'app_distributor': 'berlinphil',
+                'app_version': '1.0.0',
+                'client_secret': '2ySLN+2Fwb',
+            }), headers={
+                'Content-Type': 'application/x-www-form-urlencoded',
+            })
+        self._ACCESS_TOKEN = token_response.get('access_token')
+        try:
+            self._download_json(
+                self._OAUTH_URL,
+                None, 'Logging in', errnote='Unable to login', data=urlencode_postdata({
+                    'grant_type': 'password',
+                    'username': username,
+                    'password': password,
+                }), headers={
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                    'Referer': 'https://www.digitalconcerthall.com/en/login',
+                    'Authorization': 'Bearer ' + self._ACCESS_TOKEN
+                })
+        except ExtractorError as e:
+            msg = 'Unable to login'
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 401:
+                resp = self._parse_json(e.cause.read().decode(), None, fatal=False)
+                if resp:
+                    error = resp.get('extra_info') or resp.get('error_description') or resp.get('error')
+                    if error:
+                        msg += ': ' + error
+            raise ExtractorError('Unable to log in: ' + msg)
```

Comment:
this try-except is not really needed imo.
This a single static request which can be easily debugged so no point in being extra verbose and providing the error message from the api.